lib/models/device.coffee

```diff
@@ -389,7 +381,11 @@ getDeviceModel = (deps, opts) ->
 	# });
 	###
 	exports.getApplicationName = (uuidOrId, callback) ->
-		exports.get(uuidOrId, select: 'application_name').get('application_name').asCallback(callback)
+		exports.get uuidOrId,
+			expand: 'application': $select: 'app_name'
```

Comment:
Can this be:
```
select: id
expand: 'application': $select: 'app_name'
```
to avoid fetching the device fields we don't care about for this case (or invert the query to be on application and filter by device/(id|uuid) - I expect that's a lot harder though)

Reply:
Done (the simpler option - the 2nd is indeed much harder)